File: pubsub/src/main/java/glitch/pubsub/object/adapters/ModerationActionAdapter.java
package glitch.pubsub.object.adapters;

import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import glitch.pubsub.object.json.ModerationData;
import java.io.IOException;

public class ModerationActionAdapter extends TypeAdapter<ModerationData.Action> {
    @Override
    public void write(JsonWriter out, ModerationData.Action value) throws IOException {
        out.value(value.name().toLowerCase());
    }

    @Override
    public ModerationData.Action read(JsonReader in) throws IOException {
        return ModerationData.Action.valueOf(in.nextString().toUpperCase());
    }
}

File: pubsub/src/main/kotlin/io/glitchlib/pubsub/model/adapters/ModerationActionAdapter.kt
package io.glitchlib.pubsub.model.adapters

import com.google.gson.TypeAdapter
import com.google.gson.stream.JsonReader
import com.google.gson.stream.JsonWriter
import io.glitchlib.pubsub.model.json.ModerationData
import java.io.IOException

class ModerationActionAdapter : TypeAdapter<ModerationData.Action>() {
    @Throws(IOException::class)
    override fun write(out: JsonWriter, value: ModerationData.Action) {
        out.value(value.name.toLowerCase())
    }

    @Throws(IOException::class)
    override fun read(`in`: JsonReader): ModerationData.Action {
        return ModerationData.Action.valueOf(`in`.nextString().toUpperCase())
    }
}
